### DISCUSSION TITLE
Support custom rpc methods

### DISCUSSION THREAD

**Discussion by SonOfMosiah** [wevm/viem#231] [Idea / Feature Request] [answered]
It would be great to see support for custom rpc methods that may be unique to certain node providers. For example, a way to call `tenderly_setStorageAt` as a testClient.

The equivalent for this in Ethers is provider.send('tenderly_setStorageAt', args)

**Comment by jxom** [comment, marked as answer]
You can use `request` on the client for custom methods: 

```ts
testClient.request({ method: ‘tenderly_setStorageAt’, params: […args]
```

Though, if tenderly supports the same methods as anvil, hardhat, etc. might be worth adding support for it in the test client.

**Reply by SonOfMosiah** [reply]
This is exactly what I was looking for.

As far as Tenderly goes, I'm not sure how much overlap exists for method support, but perhaps it can be added with a subset of the available methods?
